swift/ql/lib/codeql/swift/frameworks/StandardLibrary/WebView.qll

```diff
@@ -33,6 +33,58 @@ private class WKScriptMessageBodyInheritsTaint extends TaintInheritingContent,
   }
 }
 
+/**
+ * A type or extension delcaration that adopts the protocol `WKNavigationDelegate`.
+ */
+private class AdoptsWkNavigationDelegate extends Decl {
+  AdoptsWkNavigationDelegate() {
+    exists(ProtocolDecl delegate |
+      this.(ExtensionDecl).getAProtocol().getABaseTypeDecl*() = delegate or
+      this.(NominalTypeDecl).getABaseTypeDecl*() = delegate
```

Comment:
I'm doing some work on cleaning up cases like this as it should really be a one-liner (when we have the right library support).  I'll be sure to include this case once its merged.

Reply:
Better late than never: https://github.com/github/codeql/pull/14692